src/test/rustdoc-js-std/parser-ident.js

```diff
@@ -0,0 +1,77 @@
+const QUERY = [
+    "R<!>",
+    "!",
+    "a!",
+    "a!::b",
```

Comment:
Please add a test case for `-> !`?

That seems like something that would be worth putting extra work into to make sure it work right, since diverging functions are pretty common, but also special-cased enough that it's likely to be broken if we don't test it.